roles/k3s/master/tasks/main.yml

```diff
@@ -59,15 +59,23 @@
     mode: 0644
   when: ansible_hostname == hostvars[groups['master'][0]]['ansible_hostname']
 
-- name: Init cluster inside the transient k3s-init service
-  command:
-    cmd: "systemd-run -p RestartSec=2 \
-                      -p Restart=on-failure \
-                      --unit=k3s-init \
-                      k3s server {{ server_init_args }}"
-    creates: "{{ systemd_dir }}/k3s.service"
-  args:
-    warn: false  # The ansible systemd module does not support transient units
+- block:
+    - name: Init cluster inside the transient k3s-init service
+      command:
+        cmd: "systemd-run -p RestartSec=2 \
+                          -p Restart=on-failure \
+                          --unit=k3s-init \
+                          k3s server {{ server_init_args }}"
+        creates: "{{ systemd_dir }}/k3s.service"
+      args:
+        warn: false  # The ansible systemd module does not support transient units
+  rescue:
+    - name: Attempting to rerun transient unit without p flags
+      command:
+        cmd: "systemd-run --unit=k3s-init k3s server {{ server_init_args }}"
+        creates: "{{ systemd_dir }}/k3s.service"
```

Comment:
Why are we doing this? For compatibility wither older systemd versions?

Reply:
the -p flags gave me error on CentOS 7 , removing them seemed to work for me

Reply:
IMO, before removing the flags, we should have at least answers to these to questions in order to make an informed decision:

- Why are they included in the current case?
- Why can't they be used in CentOS 7?

Reply:
sure, is your question to me or to some maintainer of the project who is also tuned in on this conversation?

Reply:
> - Why are they included in the current case?

`git blame` shows that @timothystewart6 wrote this originally. @timothystewart6, can you answer this?

---

> - Why can't they be used in CentOS 7?

@dberardo-com since you wrote this, this is a question to you :-)

Reply:
Why can't they be used in CentOS 7? --> no idea, maybe older version of systemd? dunno

Reply:
CentOS are you using? What does it print when you run man systemd ? I assume it supports -p.